### PR TITLE
Use the sriov operator to setup sr-iov in the kind sriov provider

### PR DESCRIPTION
**Pull request by fedepaol**
This PR switches the sriov provider to setup all the needed configuration using the openshift sriov operator.

In case of multiple physical functions on the host, it uses the first one it finds to configure the node.

This setup was tested on CI in https://github.com/kubevirt/kubevirt/pull/2562

### CODE REVIEW

**Comment by fedepaol** [comment]
/hold cancel

The manual run of https://github.com/kubevirt/kubevirt/pull/2562 passed (https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/2562/pull-kubevirt-e2e-kind-k8s-sriov-1.14.2/1164849130082668544)

File: cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh [hunk 1]
```diff
@@ -8,70 +8,66 @@ function wait_containers_ready {
...
-            if [ -e $virtfn/driver/unbind ]; then
-                echo $pciid > $virtfn/driver/unbind
-            fi
-            echo $(lspci -n -s $pciid | sed 's/:/ /g' | awk '{print $4 " " $5}') > /sys/bus/pci/drivers/vfio-pci/new_id
-        done
-    done
+function deploy_sriov_operator {
+  OPERATOR_PATH=${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/sriov-network-operator
+  if [[ ! -d $OPERATOR_PATH ]]; then
+    git clone https://github.com/openshift/sriov-network-operator.git $OPERATOR_PATH
```
Comment (phoracek) [inline cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh]: We should checkout specific commit. It is risky to use master, don't you think?

File: cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh [hunk 2]
```diff
@@ -8,70 +8,66 @@ function wait_containers_ready {
...
+
+sleep 5 #let the daemons appear
+SRIOVCNI_DAEMON_POD=$(kubectl get pods -n sriov-network-operator | grep sriov-cni | awk '{print $1}')
+kubectl wait --for=condition=Ready -n sriov-network-operator pod $SRIOVCNI_DAEMON_POD --timeout 3m
+
+SRIOVDEVICEPL_DAEMON_POD=$(kubectl get pods -n sriov-network-operator | grep sriov-device | awk '{print $1}')
+kubectl wait --for=condition=Ready -n sriov-network-operator pod $SRIOVDEVICEPL_DAEMON_POD --timeout 3m
+
+${CONTROL_PLANE_CMD} chmod 666 /dev/vfio/vfio
+
```
Comment (phoracek) [inline cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh]: please drop the trailing empty line

File: cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh [hunk 3]
```diff
@@ -15,7 +16,9 @@ function deploy_sriov_operator {
   fi
 
   pushd $OPERATOR_PATH
-    make deploy-setup-k8s
+    export OPERATOR_EXEC=kubectl
```
Comment (phoracek) [inline cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh]: why don't you pass it the same way you pass SHELL?

File: cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh [hunk 4]
```diff
@@ -8,70 +8,66 @@ function wait_containers_ready {
...
+deploy_sriov_operator
+
+kubectl label node sriov-control-plane node-role.kubernetes.io/worker=
+kubectl label node sriov-control-plane sriov=true 
+kubectl wait --for=condition=Ready pod --all -n sriov-network-operator --timeout 6m
+
+# fill the pf selector and the number of vfs (to prevent reboots in case of mellanox cards)
+envsubst < $MANIFESTS_DIR/network_config_policy.yaml | kubectl create -f -
+
+sleep 5 #let the daemons appear
```
Comment (booxter) [inline cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh]: this is a bit fragile b/c if it fails in 5 seconds then we'll fail the script. maybe spin a bit waiting?

Comment (booxter) [reply cluster-up/cluster/kind-k8s-sriov-1.14.2/config_sriov.sh]: Nevermind, I haven't noticed latest commit fixing that.

**Pull request merged**